test/unit/errors/error-collector.test.js

```diff
@@ -2370,3 +2370,80 @@ test('When using the async listener', (t) => {
     })
   }
 })
+
+tap.test('_processErrors', (t) => {
+  t.autoend()
+  let agent = null
+  let transaction = null
+  let errorCollector = null
+
+  t.beforeEach(() => {
+    if (agent) {
+      helper.unloadAgent(agent)
+    }
+    agent = helper.loadMockedAgent({
+      attributes: {
+        enabled: true
+      }
+    })
+    transaction = new Transaction(agent)
+    transaction.url = '/'
+    errorCollector = agent.errors
+  })
+
+  t.afterEach(() => {
+    helper.unloadAgent(agent)
+  })
```

Comment:
I'm aware that the other sub-tests in this suite follow the pattern used here, but I'd like us to avoid it. First, a little information:

1. The test runner, `tap`, runs individual test files (suites) in parallel with their own environments. That is `tap suite1.js suite2.js` will run both `suite1.js` and `suite2.js` at the same time. No variables or state defined _wholly_ within each file will interact with the other's (external resources, e.g. a database, does not have that guarantee).
2. It is _possible_ to define `tap.jobs = 2 // any integer, really` to trigger running at least 2 subtests at a time in parallel (https://tapjs.github.io/tapjs/classes/_tapjs_test.index.Test.html#jobs).

If we were to ever do 2, then the setup in this subtest will introduce race conditions on the value for `agent`, `transaction`, and `errorCollector`. Thus making it indeterminate as to what values they have at any given instant.

We can avoid such a scenario by utilizing the `t.context` object as shown in https://github.com/newrelic/node-newrelic/blob/3a910ef29c76cfd05903f01fb84d6775f8669578/test/unit/utilization/docker-info.test.js#L16-L56

Even if we never introduce parallel execution of subtests, the pattern I am suggesting gives each subtest a guaranteed (as much as can be with the mock agent, anyway) clean slate for doing its work. In effect, we will know for a fact that the result of one subtest is not affect the state of another subtest.

Reply:
Okay, that makes sense. Do you think in the future we should try to remove this pattern in the other tests in favor of the one you provided?

Reply:
If there isn't much work involved, I think it is a good idea. It's not something we need to go out of our way to do, though.